File: source/navigation/EnjoyFragmentNavigationDemo/app/src/main/java/com/example/xiangxue/enjoyfragmentnavigationdemo/fragment/TwoFragment.java
package com.example.xiangxue.enjoyfragmentnavigationdemo.fragment;

import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.xiangxue.enjoyfragmentnavigationdemo.R;

public class TwoFragment extends Fragment {


    public static Fragment newIntance() {
        TwoFragment fragment = new TwoFragment();
        return fragment;
    }

    public TwoFragment(){
        Log.i("Zero","init: " + this.getClass().getSimpleName());
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_2, container, false);
    }


    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


    }

    @Override
    public void onResume() {
        super.onResume();
        Log.i("Zero","onResume: " + this.getClass().getSimpleName());
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        Log.i("Zero","onDestroy: " + this.getClass().getSimpleName());
    }

//    public void onClick(View view) {
//
//    }


}

File: source/jetPackDemo/navigationdemo/src/main/java/com/zero/navigationdemo/basic/TwoFragment.kt
package com.zero.navigationdemo.basic

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.Navigation
import com.zero.navigationdemo.LifeCycleFragment
import com.zero.navigationdemo.R
import com.zero.navigationdemo.databinding.FragmentOneBinding
import com.zero.navigationdemo.databinding.FragmentTwoBinding

class TwoFragment: LifeCycleFragment() {

    private val binding by lazy {
        FragmentTwoBinding.inflate(layoutInflater)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding.button.setOnClickListener {
            //方式二
            Navigation.createNavigateOnClickListener(R.id.action_twoFragment_to_threeFragment).onClick(it)
        }
        return binding.root
    }

}